examples/routing_example/server/index.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { PluginInitializer } from 'kibana/server';
+
+import { RoutingExamplePlugin } from './plugin';
+
+export const plugin: PluginInitializer<void, void> = () => new RoutingExamplePlugin();
```

Comment:
will do.  There are other example plugins that use `void`.  Couldn't this be type checked?    `TSetup extends object = object,`

Reply:
Yea, the default is `void`

```ts
export interface Plugin<
  TSetup = void,
  TStart = void,
  TPluginsSetup extends object = object,
  TPluginsStart extends object = object
> {
```

which makes it actually seem like `void` is supposed to be the default, not {}.

Reply:
> There are other example plugins that use void. Couldn't this be type checked?

Yea, the current default is `void`. And even some non-example plugins are using it. 

TBH I tried a few weeks back to change it to `extends object` and adapt all usages, but this is way bigger than is looks due to the amount of plugins, and this type is using in some sensible TS magic in `core`, causing lots of TS errors.

You can keep `void` for now if you want, as we'll have to fix all the plugins at once when we'll perform this change.

@restrry @joshdover Before I open an actual issue, do we all agree that plugins should always return a contract (even empty) instead of `void`?